src/data_provider/src/sysInfoMac.cpp

```diff
@@ -102,88 +104,164 @@ nlohmann::json SysInfo::getHardware() const
+                        }
+
+                        std::string pathSubDirectory { directory + "/" + subDirectory };
+                        pkgAnalizeDirectory(pathSubDirectory);
+                    }
                 };
 
-                std::pair<SQLite::IStatement&, const int&> pkgContext {std::make_pair(std::ref(stmt), std::cref(pkgType))};
+                pkgAnalizeDirectory(pkgDirectory);
+                break;
+            }
 
-                while (SQLITE_ROW == stmt.step())
+        case RCP:
+            {
+                static auto isInPKGDirectory
                 {
-                    try
+                    [](const std::string & plistDirectory)
                     {
-                        nlohmann::json jsPackage;
-                        FactoryPackageFamilyCreator<OSPlatformType::BSDBASED>::create(pkgContext)->buildPackageData(jsPackage);
+                        for (const auto& packagesDirectory : s_mapPackagesDirectories)
+                        {
+                            if (packagesDirectory.second == RCP && Utils::startsWith(plistDirectory, packagesDirectory.first))
+                            {
+                                return false;
+                            }
+                        }
 
-                        if (!jsPackage.at("name").get_ref<const std::string&>().empty())
+                        for (const auto& packagesDirectory : s_mapPackagesDirectories)
                         {
-                            // Only return valid content packages
-                            callback(jsPackage);
+                            if (packagesDirectory.second == PKG && Utils::startsWith(plistDirectory, packagesDirectory.first))
+                            {
+                                return true;
+                            }
                         }
+
+                        return false;
                     }
-                    catch (const std::exception& e)
+                };
+
+                const auto files { Utils::enumerateDirTypeRegular(pkgDirectory) };
+
+                for (const auto& file : files)
+                {
+                    if (Utils::endsWith(file, ".plist"))
                     {
-                        std::cerr << e.what() << std::endl;
+                        std::string package { Utils::substrOnFirstOccurrence(file, ".plist") };
+
+                        try
+                        {
+                            nlohmann::json jsPackage;
+                            FactoryPackageFamilyCreator<OSPlatformType::BSDBASED>::create(std::make_pair(PackageContext{pkgDirectory, package, ""}, RCP))->buildPackageData(jsPackage);
+
+                            if (!jsPackage.at("name").get_ref<const std::string&>().empty() &&
+                                    !jsPackage.at("location").get_ref<const std::string&>().empty() &&
+                                    !isInPKGDirectory(jsPackage.at("location").get_ref<const std::string&>())
```

Comment:
We should analyze this with team.

Maybe we are adding an overhead analyzing the whole BOM file and reading a .plist twice to end up discarding the entry in `isInPKGDirectory()`? If we cant call this method before the whole process (because we don't know the location of the .plist file), couldn't we filter by package name or something?

Another alternative might be collecting some typical paths in BOM files and add them to PKG Wrapper.
For example, with **osquery**, we can get the same result adding `/opt` to the PKG directory map

![2023-08-16_00-25](https://github.com/wazuh/wazuh/assets/65046601/1a9b5c27-5414-43f9-8c2d-9eca19dab708)

Reply:
Understand your point, but one correction, you only parse the second plist file only if it is outside the PKG scanned directories.
Also (at least in a freshly installed macos) files scanned from receipts are significantly less than the ones scanned in PKG directories.
The effort to scan PKG directories is significantly greater than analyzing some RCP files.

Reply:
Ok, thank you.